api/types/filters/parse.go

```diff
@@ -262,7 +262,7 @@ func (invalidFilter) InvalidParameter() {}
 func (args Args) Validate(accepted map[string]bool) error {
 	for name := range args.fields {
 		if !accepted[name] {
-			return invalidFilter{errors.New("invalid filter '" + name + "'")}
+			return invalidFilter{errors.Errorf("invalid filter %q", name)}
```

Comment:
This changes the single quotes to double quotes; this _may_ have been done on purpose (logs / JSON would escape double quotes, which can add some additional noise).


Was there a specific reason for switching to use `Errorf` (with placeholders)? Otherwise if we just swap `errors` for `pkg/errors` (using `pkg/errors` with `errors.New()`) would still produce an error with stack-trace, without updating these lines.

Reply:
I may be wrong, but doc says:
> %q	a single-quoted character literal safely escaped with Go syntax.

Primary goal is not to get stack trace (which is nice as well) but to avoid string concatenation to produce error message.

Reply:
interesting: this doc applies to integer, but for strings:
> %q	a double-quoted string safely escaped with Go syntax

so go is definitively inconsistent. Nice to know

Reply:
Why avoid string concatenation? I realise for error messages it's not critical, but `Sprintf` definitely comes with an overhead.

Reply:
ok